app/models/goal.py

```diff
@@ -1,5 +1,24 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
 
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .task import Task
+
+
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self):
+        goal_as_dict = {
+            "id": self.id,
+            "title": self.title
+        }
+        return goal_as_dict
```

Comment:
You can directly return the dictionary literal instead of creating a variable. If you created a variable, referenced it and performed some logic on it, then I can understanding needing `goal_as_dict`, but since you don't do that, prefer:

```suggestion
        return {
            "id": self.id,
            "title": self.title
        }
```